pkg/kubelet/network/hostport/hostport_manager.go

```diff
@@ -152,6 +156,12 @@ func (hm *hostportManager) Add(id string, podPortMapping *PodPortMapping, natInt
 		// clean up opened host port if encounter any error
 		return utilerrors.NewAggregate([]error{err, hm.closeHostports(hostportMappings)})
 	}
+
+	// Ensure the chain is linked from the root.
+	if err = ensureKubeHostportChainLinked(hm.iptables); err != nil {
```

Comment:
I think this needs to be at the top. Say if `ensureKubeHostportChainLinked` failed during bootstrap in `NewHostpoortManager`, KUBE-HOSTPORTS chain will not be created. Then `iptables-restore` above will always fail, and never able to reach here.